src/plugins/intel_gpu/src/graph/impls/cm/vl_sdpa_opt.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "vl_sdpa_opt.hpp"
+
+#include <algorithm>
+#include <cmath>
+
+#include "common_utils/kernel_generator_base.hpp"
+#include "intel_gpu/primitives/vl_sdpa.hpp"
+#include "primitive_cm_base.hpp"
+#include "primitive_inst.h"
+#include "registry/implementation_manager.hpp"
+#include "utils/kernel_generator.hpp"
+
+namespace ov::intel_gpu::cm {
+namespace {
+
+constexpr auto get_vlsdpa_build_options() {
+    return " -cmc -Qxcm_register_file_size=256";
+}
+
+class VLSDPAGenerator : public KernelGenerator {
+public:
+    VLSDPAGenerator() : KernelGenerator("cm_sdpa_vlen") {}
+
+protected:
+    [[nodiscard]] std::string get_build_options(const RuntimeParams& params) const override {
+        return KernelGenerator::get_build_options(params) + get_vlsdpa_build_options();
+    }
+
+    [[nodiscard]] JitConstants get_jit_constants(const RuntimeParams& params) const override {
+        auto jit = KernelGenerator::get_jit_constants(params);
+
+        // transpose shape into BHLS(4D), or HLS(3D)
+        auto transpose_pshape = [](const ov::PartialShape& pshape, const std::vector<int64_t>& order) {
+            if (order.empty())
+                return pshape;
+
+            auto transposed_pshape = ov::PartialShape::dynamic(pshape.rank());
+            for (size_t i = 0; i < order.size(); i++) {
+                transposed_pshape[i] = pshape[order[i]];
+            }
+            return transposed_pshape;
+        };
+
+        auto desc = params.typed_desc<vl_sdpa>();
+        const auto query_shape = transpose_pshape(params.get_input_layout(0).get_partial_shape(), desc->input_q_transpose_order);
+        const auto key_shape = transpose_pshape(params.get_input_layout(1).get_partial_shape(), desc->input_k_transpose_order);
+
+        const size_t head_size = key_shape[query_shape.size() - 1].get_length();
+        const size_t num_q_heads = query_shape[query_shape.size() - 3].get_length();
+        const size_t num_kv_heads = key_shape[key_shape.size() - 3].get_length();
+        const float scale_factor = 1.0 / std::sqrt(static_cast<double>(head_size));
```

Comment:
@sshlyapn Is there any good place that I can keep them as we did in get_sdpa_configuration() of src/plugins/intel_gpu/src/graph/impls/ocl/scaled_dot_product_attention.cpp? Then I can reuse them directly later.

Reply:
Hmm, code generation is independent of the parameters, but maybe it would be more convenient to at least add some custom helper structure?
```
struct VLSDPAParams {
    VLSDPAParams(const RuntimeParams& params) { /* calculate and parse parameters */ }

    size_t head_size;
    size_t num_q_heads;
    ...
}
```

Reply:
@riverlijunjie how do you think? I bet you meet the same problem in task of PA/SDPA new infra update.